pkg/collector/corechecks/containers/kubernetes_apiserver.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 Datadog, Inc.
+
+// +build kubeapiserver
+
+package containers
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	log "github.com/cihub/seelog"
+	yaml "gopkg.in/yaml.v2"
+	"time"
+	//"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"context"
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/ericchiang/k8s"
+)
+
+// Covers the Controle pannel service check and the in memory pod metadata.
+
+const kubernetesApiServerCheck = "kubernetes_apiserver"
+
+const (
+	KubeControlPaneCheck string = "kubernetes_APIServer.controlePlaneUp"
+)
+
+type KubeASConfig struct {
+	//CollectContainerSize bool               `yaml:"collect_container_size"`
+	Tags []string `yaml:"tags"`
+	//CollectEvent         bool               `yaml:"collect_events"`
+	//FilteredEventType    []string           `yaml:"filtered_event_types"`
+}
+
+// KubeASCheck grabs docker metrics
+type KubeASCheck struct {
+	lastWarnings []error
+	instance     *KubeASConfig
+	//lastEventTime  time.Time
+	KubeApiServerHostname string
+}
+
+func (k *KubeASCheck) String() string {
+	return kubernetesApiServerCheck
+}
+
+func (c *KubeASConfig) Parse(data []byte) error {
+
+	if err := yaml.Unmarshal(data, c); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Configure parses the check configuration and init the check
+func (k *KubeASCheck) Configure(config, initConfig check.ConfigData) error {
+	k.instance.Parse(config)
+	// Check connectivity to the APIServer
+	log.Infof("Running config %s", config)
+	return nil
+}
+
+// GetWarnings grabs the last warnings from the sender
+func (k *KubeASCheck) GetWarnings() []error {
+	w := k.lastWarnings
+	k.lastWarnings = []error{}
+	return w
+}
+
+// ID returns the name of the check since there should be only one instance running
+func (k *KubeASCheck) ID() check.ID {
+	return check.ID(k.String())
+}
+
+// Interval returns the scheduling time for the check
+func (k *KubeASCheck) Interval() time.Duration {
+	return check.DefaultCheckInterval
+}
+
+// Run executes the check
+func (k *KubeASCheck) Run() error {
+	sender, err := aggregator.GetSender(k.ID())
+	if err != nil {
+		return err
+	}
+	client, err := k8s.NewInClusterClient()
+	if err != nil {
+		log.Errorf("could not instantiate the cluster: %q", err)
+		return nil
+	}
+	componentsStatus, err := client.CoreV1().ListComponentStatuses(context.Background())
+	if err != nil {
+		log.Errorf("could not retrieve the status from the control pannel's components", err)
+		return nil
+	}
+	for _, component := range componentsStatus.Items {
+
+		tag_comp := []string{}
+		tag_comp = append(k.instance.Tags, *component.Metadata.Name)
+
+		// We only expect the Healthy condition. May change in the future. https://godoc.org/github.com/ericchiang/k8s/api/v1#ComponentCondition
+		for condition := range component.Conditions {
```

Comment:
If you write `for _, condition`, `condition` will hold the object instead of the index, you won't have to do the lookup after